#153756400 Convert from grunt to Gulp taskrunner

kevinejiro opened the pull request:
### What does this PR do?
Converts from grunt to gulp task runner

### Description of Task to be completed?
Grunt is old; with Gulp, the syntax is much simpler and it uses pipe streams which makes it inherently faster.

### How should this be manually tested?
Run `npm install` (To Install all dependencies )
Run `npm start` (To start gulp)

### Any background context you want to provide?
N/A

### What are the relevant pivotal tracker stories?
153756400

### Screenshots (if appropriate)
<img width="1163" alt="screen shot 2018-01-03 at 5 22 26 pm" src="https://user-images.githubusercontent.com/28443813/34529273-b426ef64-f0ab-11e7-9f38-deb3190414d9.png">


### Questions:
N/A

kevinejiro committed:
chore (Gulp): Convert from Grunt to Gulp taskrunner

- delete gruntfile.js
- delete grunt dependencies from package.json
- add gulpfile.js
- add gulp to dev dependencies
- add gulp-sass to dev dependencies
- add browser-sync to dev dependencies
- add gulp-bower to dev dependencies
- add gulp-eslint to dev dependencies
- add gulp-mocha to dev dependencies
- add gulp-exit to dev dependencies
- add gulp-nodemon to dev dependencies
- update start, test, and postinstall scripts in package.json
- update server.js
- add bower_components/ and  .nyc_output/ to .gitignore file
- update common.css

[Finishes #153756400]

temilaj commented on package.json:
```diff
@@ -13,25 +13,23 @@
-    "start": "node node_modules/grunt-cli/bin/grunt",
-    "test": "node node_modules/grunt-cli/bin/grunt test",
-    "postinstall": "node node_modules/grunt-cli/bin/grunt install",
+    "start": "node node_modules/gulp/bin/gulp build && node node_modules/gulp/bin/gulp",
```
why do we have to run gulp twice?

kevinejiro replied:
I did that because we might have to refactor our code to es6 which might give problems on heroku.

temilaj reviewed (changes requested):
made a comment. please review

kevinejiro committed:
chore (Gulp): Convert from Grunt to
Gulp taskrunner

- add heroku-postinstall script to package.json
- update start script
- add start:dev script to package.json
- update gulpfile.js

[Finishes #153756400]

temilaj reviewed (changes requested):
Both your commits seem to be doing the same thing. Please squash them into a single commit. Or have separate commits for the different tasks completed
